mypy/types.py

```diff
@@ -867,6 +867,53 @@ def deserialize(cls, data: JsonDict) -> 'AnyType':
                        AnyType.deserialize(source) if source is not None else None,
                        data['missing_import_name'])
 
+    def describe(self) -> str:
+        if not mypy.options._based:
+            return 'Any'
+
+        def type_of_any(toa: int = self.type_of_any) -> Optional[str]:
+            if toa == TypeOfAny.unannotated:
+                return "unannotated"
+            elif toa == TypeOfAny.explicit:
+                return None
+            elif toa == TypeOfAny.from_unimported_type:
+                return "from unimported type"
+            elif toa == TypeOfAny.from_omitted_generics:
+                return "from omitted generics"
+            elif toa == TypeOfAny.from_error:
+                return "from error"
+            elif toa == TypeOfAny.special_form:
+                return "special form"
+            elif toa == TypeOfAny.from_another_any:
+                return type_of_any(self.source_any.type_of_any) if self.source_any else None
+            elif toa == TypeOfAny.implementation_artifact:
+                return "from a limitation"
+            elif toa == TypeOfAny.suggestion_engine:
+                return "from a suggestion"
```

Comment:
i think its sus to show stuff to the user when even we dont know what it means

Reply:
It's barely used, and is a heuristic I think, so I vote it's okay.